kong/clustering/compat/init.lua

```diff
@@ -0,0 +1,335 @@
+local constants = require("kong.constants")
+local meta = require("kong.meta")
+local version = require("kong.clustering.compat.version")
+local utils = require("kong.tools.utils")
+
+local type = type
+local ipairs = ipairs
+local table_insert = table.insert
+local table_sort = table.sort
+local gsub = string.gsub
+local deep_copy = utils.deep_copy
+local split = utils.split
+
+local ngx = ngx
+local ngx_log = ngx.log
+local ngx_INFO = ngx.INFO
+local ngx_NOTICE = ngx.NOTICE
+local ngx_WARN = ngx.WARN
+
+local version_num = version.string_to_number
+local extract_major_minor = version.extract_major_minor
+
+local _log_prefix = "[clustering] "
+
+local REMOVED_FIELDS = require("kong.clustering.compat.removed_fields")
+local CLUSTERING_SYNC_STATUS = constants.CLUSTERING_SYNC_STATUS
+local KONG_VERSION = meta.version
+
+
+local _M = {}
+
+
+local function check_kong_version_compatibility(cp_version, dp_version, log_suffix)
```

Comment:
I think you should add a test for this to `03-compat_spec.lua` so we know if someone accidentally changes this behavior

Reply:
This is not a good idea, but I think it's okay to go without more unit tests for this at the moment.

Reason being that this is not a new function (it was moved over as-is from `kong.clustering.utils`), _and_ it has some decent integration test coverage [here](https://github.com/Kong/kong/blob/acd48732ce67ee9b2af41a482cb8b75543fae877/spec/02-integration/09-hybrid_mode/01-sync_spec.lua#L409-L420).